File: pages/api/tripadvisor_api.py
import requests

# Read API KEY from .env.local file in root folder
with open(".env.local", "r") as f:
    for line in f:
        if "TRIPADVISOR_KEY" in line:
            TRIPADVISOR_API_KEY = line.split("=")[1].strip()


search_query = "singapore"
url = f"https://api.content.tripadvisor.com/api/v1/location/search?key={TRIPADVISOR_API_KEY}&searchQuery={search_query}&language=en"

headers = {"accept": "application/json"}

response = requests.get(url, headers=headers)

print(response.text)

File: pages/api/tripadvisor_api.js
const fetch = require("node-fetch");

const test = require("dotenv").config({ path: ".env.local" });
const searchQuery = "singapore";

const url = `https://api.content.tripadvisor.com/api/v1/location/search?key=${process.env.TRIPADVISOR_KEY}&searchQuery=${searchQuery}&language=en`;
const options = { method: "GET", headers: { accept: "application/json" } };
console.log(url);

fetch(url, options)
  .then((res) => res.json())
  .then((json) => console.log(json))
  .catch((err) => console.error("error:" + err));
